Tests/IntegrationTests/PostgresNIOTests.swift

```diff
@@ -227,6 +227,23 @@ final class PostgresNIOTests: XCTestCase {
         XCTAssertNoThrow(_ = try conn?.simpleQuery("NOTIFY undesired").wait())
         XCTAssertNoThrow(_ = try conn?.simpleQuery("SELECT 1").wait())
     }
+    
+    func testNotificationWithKeywordChannelName() {
+        var conn: PostgresConnection?
+        XCTAssertNoThrow(conn = try PostgresConnection.test(on: eventLoop).wait())
+        defer { XCTAssertNoThrow( try conn?.close().wait() ) }
+
+        var receivedNotifications: [PostgresMessage.NotificationResponse] = []
+        conn?.addListener(channel: "default") { context, notification in
+            receivedNotifications.append(notification)
+        }
+        XCTAssertNoThrow(_ = try conn?.simpleQuery("LISTEN \"default\"").wait())
+        XCTAssertNoThrow(_ = try conn?.simpleQuery("NOTIFY \"default\"").wait())
+        XCTAssertNoThrow(_ = try conn?.simpleQuery("SELECT 1").wait())
+        XCTAssertEqual(receivedNotifications.count, 1)
+        XCTAssertEqual(receivedNotifications.first?.channel, "default")
+        XCTAssertEqual(receivedNotifications.first?.payload, "")
+    }
```

Comment:
How does this test interact with the code that you changed? Maybe we want to change `AsyncPostgresConnectionTests.testListenAndNotify` to something like this:

```swift
    func testListenAndNotify() async throws {
        let channelNames = [
            "foo",
            "default"
        ]

        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
        defer { XCTAssertNoThrow(try eventLoopGroup.syncShutdownGracefully()) }
        let eventLoop = eventLoopGroup.next()

        for channelName in channelNames {
            try await self.withTestConnection(on: eventLoop) { connection in
                let stream = try await connection.listen(channelName)
                var iterator = stream.makeAsyncIterator()

                try await self.withTestConnection(on: eventLoop) { other in
                    try await other.query(#"NOTIFY "\#(unescaped: channelName)", 'bar';"#, logger: .psqlTest)

                    try await other.query(#"NOTIFY "\#(unescaped: channelName)", 'foo';"#, logger: .psqlTest)
                }

                let first = try await iterator.next()
                XCTAssertEqual(first?.payload, "bar")

                let second = try await iterator.next()
                XCTAssertEqual(second?.payload, "foo")
            }
        }
    }
```

Reply:
My change was related to the quoting that occurs in the `LISTEN` SQL query caused by `addListener`. So before the changes, line 237 of this new test would fail on the PostgreSQL side when running `LISTEN default;` with the error:

> ERROR:  syntax error at or near "default" at character 8
> STATEMENT:  LISTEN default;

Reply:
I'm totally happy to combine it into `testListenAndNotify` though if you prefer it that way.

Reply:
Cool, I've made your recommended changes to `testListenAndNotify` and removed the `testNotificationWithKeywordChannelName` test